apps/deepl_integration/handlers.py

```diff
@@ -320,34 +320,225 @@ def _process_model_preds(cls, model_version, current_tags_map, draft_entry, mode
 
     @classmethod
     def save_data(cls, draft_entry, data):
-        model_preds = data['model_preds']
+        model_preds = data
         # Save if new tags are provided
         current_tags_map = cls._get_or_create_tags_map([
             tag
-            for prediction in model_preds
-            for category_tag, tags in prediction.get('tags', {}).items()
+            for category_tag, tags in model_preds['model_tags'].items()
             for tag in [
                 category_tag,
                 *tags.keys(),
             ]
         ])
-        models_version_map = cls._get_or_create_models_version([
-            prediction['model_info']
-            for prediction in model_preds
-        ])
-
+        models_version_map = cls._get_or_create_models_version(
+            [
+                model_preds['model_info']
+            ]
+        )
         with transaction.atomic():
             draft_entry.clear_data()  # Clear old data if exists
             draft_entry.calculated_at = timezone.now()
-            for prediction in model_preds:
-                model_version = models_version_map[(prediction['model_info']['id'], prediction['model_info']['version'])]
-                cls._process_model_preds(model_version, current_tags_map, draft_entry, prediction)
+            model_version = models_version_map[(model_preds['model_info']['id'], model_preds['model_info']['version'])]
+            cls._process_model_preds(model_version, current_tags_map, draft_entry, model_preds)
             draft_entry.prediction_status = DraftEntry.PredictionStatus.DONE
             draft_entry.save_geo_data()
             draft_entry.save()
         return draft_entry
 
 
+class AutoAssistedTaggingDraftEntryHandler(BaseHandler):
+    model = Lead
```

Comment:
Let's add a TODO here.
```suggestion
class AutoAssistedTaggingDraftEntryHandler(BaseHandler):
    # TODO: Fix N+1 issues here. Try to do bulk_update for each models.
    model = Lead
```